airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/source.py

```diff
@@ -83,10 +90,21 @@ def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
             - json_schema providing the specifications of expected schema for this stream (a list of columns described
             by their names and types)
         """
+        dataset_client = self._get_dataset_client(config)
+
+        # Get total number of items in dataset. This will be used in pagination
+        dataset = dataset_client.get()
+        fields = dataset["fields"]
         stream_name = DATASET_ITEMS_STREAM_NAME
         json_schema = {
             "$schema": "http://json-schema.org/draft-07/schema#",
             "type": "object",
+            "properties": {
+                # as datasets are not typed and we only know the field name, each field is defined as oneOf of all possible types
+                field: {"type": ["array", "object", "boolean", "number", "integer", "string"]}
```

Comment:
I think this will confuse downstream destinations. Should we just embed all the data in a wrapper object whose fields aren't declared?

Reply:
AFAIK it should turn every column into a jsonb which seemed better to me than a single wrapper object.
I actually started out that way (see the first commit https://github.com/airbytehq/airbyte/pull/28290/commits/03ceb1de88afff54e6f2c5682063d64d43c8c6b3 ) but I reconsidered because while not perfect this approach seemed a little better.

If you have concerns we can also go with the wrapper object for now (even though it would be a breaking change but that's probably not a huge deak as it's not functional at all at the moment)

Reply:
I think I'd recommend a wrapper object since it's get translated as `JSONB` or `VARIANT` etc.. by most downstream destinations